### ISSUE TITLE
BCD "category" can be derived from the query

### ISSUE THREAD

**Issue by peterbe** [mdn/stumptown-renderer#188]
See: https://github.com/mdn/stumptown-renderer/blob/9a95f4a88d4840749b846b44401be8558911c9d4/client/src/ingredients/browser-compatibility-table/index.js#L82

The component is never called with a `category` prop. 
However, as of yesterday, we do have `data.query` as a prop. That is the key to *derive* a different value to the `category` variable. 

According to @Elchi3 that value should be the first part of the query. So if the query is `'html.elements.video'` the category should be `'html'`. I.e. `data.query.split('.').shift()`.

**Comment by peterbe** [comment]
Caveat: We don't have any stumptown-content whose `data.query` does NOT start with `'html'` so to test this out you'd need to use some imagination :)

**Comment by Elchi3** [comment]
In the KumaScript implementation, category is used to decide which browsers to display in a compat table.
- By default this is all desktop and all mobile browsers (in two groups)
- if the category is 'javascript', a third group "servers" is added and nodejs appears as an additional browser.
- if the category is 'webextensions', only a small set of desktop and mobile browsers is shown, as webextension aren't in all browsers. 

I don't know if we need this in stumptown. Maybe we do, but lacking other content than "html", this needs indeed imagination.

**Comment by peterbe** [comment]
I was hoping this issue could be about messing with that parameter so it's not only hardcoded but actually depends the query. Alone. 

The actual meat about which columns to show is here: https://github.com/mdn/stumptown-renderer/blob/9a95f4a88d4840749b846b44401be8558911c9d4/client/src/ingredients/browser-compatibility-table/index.js#L35-L50
@Elchi3 Would you mind reviewing that on its own to make sure it's exactly the way you want it? ...separate from how the *argument* to that function works.

**Comment by Elchi3** [comment]
> @Elchi3 Would you mind reviewing that on its own to make sure it's exactly the way you want it? ...separate from how the argument to that function works.

The `gatherPlatformsAndBrowsers` function does exactly do what I want, yes. What I don't know is whether we will have WebExtensions docs in stumptown, since it is rather Firefox docs than Web docs. With that being gone potentially, the logic could just be: when there is nodejs compat data present, show a third column with "Server", otherwise omit that column, and then there would be no need anymore for categorization at all.

**Comment by peterbe** [comment]
I think we should leave it as is then. And even keep the `webextensions` stuff a little bit longer because it's possible we might combine legacy Kuma docs with stumptown-renderer so then it could potentially be useful. (We'll talk more about it in London).

The next action is to actually use the first part of the query. Sadly, we don't yet have any nodejs pages to try this out. We might just do it anyway.

**Comment by joedarc** [comment]
Just to put a little update on this. Using the query for the category works as you suggested @peterbe. Ended up using it when working with the legacy Kuma docs. I'll make a PR for it at some point today.

> What I don't know is whether we will have WebExtensions docs in stumptown

Interesting, I actually ran into some discrepancies when rendering the BCD table for a `webextensions` doc. Basically, the data was a bit different than what the renderer is currently expecting. I sort of expected this as we were only working with `html` docs at the time and figured there may be some other cases down the road. Probably not relevant information for this issue specifically, just putting that in the air.
